lib/bandit/http.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Bandit.HTTP do
+  @moduledoc false
+  # Implements functions shared by different HTTP versions
+
+  @doc """
+  Checks if a header tuple list already contains a date header
+  """
+  def has_date_header?(headers) do
+    Enum.any?(headers, fn {header, _value} ->
+      String.downcase(header) == "date"
+    end)
```

Comment:
This may end up being quite expensive. Can we use an approach like https://hexdocs.pm/plug/Plug.Conn.html#get_resp_header/2 instead (note that we can rely on the `header` being lowercase).

Reply:
I updated the `has_date_header?/1` to only check for lowercase date headers.